app/main/views.py

```diff
@@ -99,6 +165,43 @@ def get_service(service_id):
     return jsonify(services=jsonify_service(service))
 
 
+@main.route('/services-archive/<int:service_archive_id>', methods=['GET'])
```

Comment:
Have you considered something like:
- `/service/<service_id>/at/<change_id>` or `/service/<service_id>/archive/<change_id>`
- and `/service/<service_id>/archives?

Don't know if that fits with the API/REST conventions but feels cleaner to me.